ibm_platform_services/ibm_cloud_shell_v1.py

```diff
@@ -0,0 +1,524 @@
+            )
+        service.configure_service(service_name)
+        return service
+
+    def __init__(self,
+                 authenticator: Authenticator = None,
+                ) -> None:
+        """
+        Construct a new client for the IBM Cloud Shell service.
+
+        :param Authenticator authenticator: The authenticator specifies the authentication mechanism.
+               Get up to date information from https://github.com/IBM/python-sdk-core/blob/master/README.md
+               about initializing the authenticator of your choice.
+        """
+        BaseService.__init__(self,
+                             service_url=self.DEFAULT_SERVICE_URL,
+                             authenticator=authenticator)
+
+
+    #########################
+    # account_settings
+    #########################
+
+
+    def get_account_settings_by_id(self,
+        account_id: str,
+        **kwargs
+    ) -> DetailedResponse:
+        """
+        Get account settings.
+
+        Retrieve account settings for the given account ID. Call this method to get
+        details about a particular account setting, whether Cloud Shell is enabled, the
+        list of enabled regions and the list of enabled features. Users need to be an
+        account owner or users need to be assigned an IAM policy with the Administrator
+        role for the Cloud Shell account management service.
+
+        :param str account_id: The account ID in which the account settings belong
+               to.
+        :param dict headers: A `dict` containing the request headers
+        :return: A `DetailedResponse` containing the result, headers and HTTP status code.
+        :rtype: DetailedResponse with `dict` result representing a `AccountSettings` object
+        """
+
+        if account_id is None:
+            raise ValueError('account_id must be provided')
+        headers = {}
+        sdk_headers = get_sdk_headers(service_name=self.DEFAULT_SERVICE_NAME,
+                                      service_version='V1',
+                                      operation_id='get_account_settings_by_id')
+        headers.update(sdk_headers)
+
+        if 'headers' in kwargs:
+            headers.update(kwargs.get('headers'))
+        headers['Accept'] = 'application/json'
+
+        path_param_keys = ['account_id']
+        path_param_values = self.encode_path_vars(account_id)
+        path_param_dict = dict(zip(path_param_keys, path_param_values))
+        url = '/api/v1/user/accounts/{account_id}/settings'.format(**path_param_dict)
+        request = self.prepare_request(method='GET',
+                                       url=url,
+                                       headers=headers)
+
+        response = self.send(request)
+        return response
+
+
+    def update_account_settings_by_id(self,
+        account_id: str,
+        *,
+        new_id: str = None,
+        new_rev: str = None,
+        new_account_id: str = None,
+        new_created_at: int = None,
```

Comment:
Defining the "created_*" and "updated_*" properties as readonly will remove them from this arg list, which is what we want (I think) :)

Reply:
Done